### PR TITLE
Add mock service worker to login sequence

### LINKED ISSUE #13: Use mock service worker
- add msw to replicate api calls: https://mswjs.io

### PR DESCRIPTION
**Pull request by borangoher**
close #13

### CODE REVIEW

File: src/mocks/handlers.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { rest } from "msw";
+
+export const handlers = [
+  rest.post("http://localhost:8080/login", async (req, res, ctx) => {
+    console.log(req);
+    await new Promise((r) => setTimeout(r, 1000)); //simulate API response delay
```
Comment (yeged) [inline src/mocks/handlers.js:6]: I think you can use `ctx.delay(1000)`

File: src/Components/login/Login.js [hunk 2]
```diff
@@ -26,7 +42,15 @@ const Login = () => {
...
-  const onSubmit = () => dispatch({ type: LoginActions.LOGIN });
+
+  const onSubmit = (userData) => {
+    //console.log(userData);
+    loginUser(userData)
+      .then(() => dispatch({ type: LoginActions.LOGIN }))
+      .catch((error) => {
+        console.log(error);
+      });
+  };
```
Comment (yeged) [inline src/Components/login/Login.js:53]: - Remove console.log
- We usually use async await. You can try 'the try catch statement'

**Review by yeged — APPROVED** [review]

File: src/mocks/handlers.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import { rest } from "msw";
+
+export const handlers = [
+  rest.post("http://localhost:8080/login", async (req, res, ctx) => {
+    console.log(req);
```
Comment (ozgunbal) [inline src/mocks/handlers.js:5]: better to remove.

File: src/Components/login/Login.js [hunk 4]
```diff
@@ -7,12 +7,28 @@ import {
...
 } from "@mui/material";
-import { useLogin, LoginActions } from "./LoginContext";
+import { useLogin, LoginActions } from "../LoginContext";
 import { useForm, Controller } from "react-hook-form";
 import { yupResolver } from "@hookform/resolvers/yup";
 import { schema } from "./Login.validation";
 import { loginDefaultValues } from "./Login.constant";
 
+const loginUser = async (userData) => {
+  const res = await fetch("http://localhost:8080/login", {
```
Comment (ozgunbal) [inline src/Components/login/Login.js:17]: using relative path would be better, then you don't need to change domain when needed.

**Review by ozgunbal — COMMENTED** [review]
You can send these items with #21

**Pull request merged**